db/deploy/mutations/parse_gis_data.sql

```diff
@@ -22,20 +23,27 @@ begin
     -- Loop over each object in the JSON array
     for json_row in select * from jsonb_array_elements(json_file)
     loop
-        ccbc_id := json_row->>'ccbc_number'; 
+        ccbc_id := json_row->>'ccbc_number';
 
         -- Look up the application_id based on the ccbc_number value
-        select id into application_row_id from ccbc_public.application where ccbc_number = ccbc_id; 
-        -- If no row was found, add to list 
-        if not found then
-            result := coalesce(result || ',','') || ccbc_id;  
-            -- RAISE EXCEPTION 'No matching row in ccbc_public.application for ccbc_number %', ccbc_id;
-        else 
-            insert into ccbc_public.application_gis_data (batch_id, application_id, json_data) 
+        select id into application_row_id from ccbc_public.application where ccbc_number = ccbc_id;
+
+        -- Look up the application_status based on the application_id value
+        select status into application_status_name from ccbc_public.application_status
+        where application_id = application_row_id
+        order by id desc limit 1;
+
+        -- If no row was found, add to list
+        if not found or
+            (application_status_name != 'received' and application_status_name != 'screening' and application_status_name != 'assessment')
```

Comment:
nit: you can add in a `application_status_name not in ('received', 'screening', 'assessment')` I think. But more of a style choice, might make things easier when we want to add in other statuses

Reply:
Thanks, I definitely think this is simpler. Updated!